active/0000-addon-black-and-whitelist-for-apps.md

```diff
@@ -0,0 +1,43 @@
+- Start Date: 2015-11-11
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+It should be possible to white- and/or blacklist addons in `EmberApp`.
+
+# Motivation
+
+If there are two (or more) `EmberApp`s, it's very likely that not all applications need all addons.
+E.g. if there is a main page application and one application for embeddable widgets, the main page might need all sorts of addons like `ember-modal-dialog` which adds completely useless bytes to widgets javascript and css files for the widgets. Other addons may add useless initializers or other things that have runtime performance penalties for no benefit.
+
+# Detailed design
+
+When EmberApp ctor gets passed a blacklist like this
+
+```javascript
+  EmberApp({
+    addonBlacklist: ['ember-modal-dialog']
+  });
+```
+
+it won't add addons whose `name` matches `ember-modal-dialog` to the list of addons for this app, just as if the addon's `isEnabled()` hook returned `false`.
+
+C.f. https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L344, this is also were I would add this check.
+
+Whitelist could work analogously.
+
+# Drawbacks
+
+- It adds a bit of API surface while you (possibly) don't care for the multiple app use case of ember-cli.
+
+- It kinda makes the `name` of an addon public api, so people might change it, not noticing they are breaking people's build (and people might not notice it either). Some addons have names like `Ember CLI ic-ajax` which seems awkward to use as an identifier.
+
+# Alternatives
+
+- Add a unified way to enable/disable an addon via normal config
```

Comment:
In fact, that is already possible today - and I considered doing it that way (by just putting two complete projects into one repository).
But I decided to not go down that road, it's a nightmare to maintain two package.json files, always having to remember to bump versions in both files. I'd say two versions of the same addon is a corner case in which it might make sense to do that. Noone stops anyone from doing that, but it would be nice for me not having to do that :-)

Still want me to add it to the RFC text as another possible option?